File: src/test/java/test/UpdateOrganizationTest.java
package test;
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class UpdateOrganizationTest {

    private static final String ACCESS_KEY_ID = "AUTHING_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "AUTHING_ACCESS_KEY_SECRET";

    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions();
        clientOptions.setAccessKeyId(ACCESS_KEY_ID);
        clientOptions.setAccessKeySecret(ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        UpdateOrganizationReqDto request = new UpdateOrganizationReqDto();
        request.setOrganizationCode("organizationCode_2401");
        request.setDescription("description_3857");
        request.setOpenDepartmentId("openDepartmentId_4592");
        request.setLeaderUserIds(Collections.singletonList("String_564"));
        request.setI18n(new cn.authing.sdk.java.dto.OrganizationNameI18nDto());
        request.setOrganizationNewCode("organizationNewCode_9569");
        request.setOrganizationName("organizationName_1160");

        OrganizationSingleRespDto response = client.updateOrganization(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/UpdateOrganizationTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.UpdateOrganizationReqDto{
        OrganizationCode: "organizationCode_7133",
        Description: "description_2722",
        OpenDepartmentId: "openDepartmentId_3745",
        LeaderUserIds: nil,
        I18n: nil,
        OrganizationNewCode: "organizationNewCode_8142",
        OrganizationName: "organizationName_4084",
    }
    response := client.UpdateOrganization(&request)
    fmt.Println(response)
}

